test/unit/v1.0.0/converter-v1-to-v2.test.js

```diff
@@ -1643,4 +1697,110 @@ describe('v1.0.0 to v2.0.0', function () {
+                    {id: undefined, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: false, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: '', responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: 0, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]}
+                ]
+            }, options, function (err, result) {
+                expect(err).to.not.be.ok;
+                expect(result && result.info).to.be.ok;
+
+                expect(result.info).to.have.property('_postman_id', '2509a94e-eca1-43ca-a8aa-0e200636764f');
+                expect(result.item).to.have.length(12);
+
+                _.forEach(result.item, function (elem) {
+                    _.forEach(elem.item, function (item) {
+                        expect(item._postman_id).to.match(/[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}/);
+                        _.forEach(item.response, function (response) {
+                            expect(response.id).to.match(/[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}/);
+                        });
+                    });
+                    expect(elem._postman_id).to.match(/[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}/);
+                });
+            });
+        });
+
+        it('should handle IDs correctly when set to false (collection id should be retained)', function () {
+            transformer.convert({
+                id: '2509a94e-eca1-43ca-a8aa-0e200636764f',
+                order: [null, NaN, undefined, false, '', 0, 'R1'],
+                folders_order: [null, NaN, undefined, false, '', 0],
+                folders: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}],
+                requests: [
+                    {id: null, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: NaN, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    // eslint-disable-next-line max-len
+                    {id: undefined, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: false, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: '', responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]},
+                    {id: 0, responses: [{id: null}, {id: NaN}, {id: undefined}, {id: false}, {id: ''}, {id: 0}]}
+                ]
+            }, _.defaults({retainIds: false}, options), function (err, result) {
+                expect(err).to.not.be.ok;
+                expect(result && result.info).to.be.ok;
+
+                // truthy collection ids should be left as is even when retainIds is set to false
```

Comment:
Just to make the comments and the test descriptions clearer, collection ids are recreated when `retainIds` is set to false. 

The only difference is `_postman_id` is always retained no matter what the flag is.